halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/services/v1/ArtifactService.java

```diff
@@ -86,10 +89,70 @@ public String getArtifactVersion(String deploymentName, SpinnakerArtifact artifa
     return getBillOfMaterials(deploymentName).getServices().getArtifactVersion(artifact);
   }
 
-  public void writeArtifactConfig(SpinnakerArtifact artifact, String profileName, String contents) {
+  public void writeBom(String bomPath) {
     if (writeableProfileRegistry == null) {
       throw new HalException(new ConfigProblemBuilder(FATAL,
           "You need to set the \"spinnaker.config.input.writer\" property to \"true\" to modify base-profiles.").build());
     }
+
+    BillOfMaterials bom;
+    String bomContents;
+    String version;
+
+    try {
+      bomContents = IOUtils.toString(new FileInputStream(bomPath));
+      bom = strictObjectMapper.convertValue(
```

Comment:
This is following the style in the rest of the repo (where I kick long parameters to the next line before chained method calls).